lib/dao.js

```diff
@@ -46,109 +38,156 @@ export class Organization {
+    /**
+     * TODO: dao.votingMachine may soon have to go
+     */
+    dao.votingMachine = await AbsoluteVote.at(defaultVotingMachineParams.votingMachineAddress);
 
-    org.votingMachine = await AbsoluteVote.at(options.votingMachine);
-    await org.votingMachine.setParameters(
-      org.reputation.address,
-      options.votePrec,
-      options.ownerVote
+    await dao.votingMachine.setParameters(
+      defaultVotingMachineParams.reputationAddress,
+      defaultVotingMachineParams.votePerc,
+      defaultVotingMachineParams.ownerVote
     );
 
-    const voteParametersHash = await org.votingMachine.getParametersHash(
-      org.reputation.address,
-      options.votePrec,
-      options.ownerVote
+    const defaultVoteParametersHash = await dao.votingMachine.getParametersHash(
+      defaultVotingMachineParams.reputationAddress,
+      defaultVotingMachineParams.votePerc,
+      defaultVotingMachineParams.ownerVote
     );
 
     // TODO: these are specific configuration options that should be settable in the options above
     const initialSchemesSchemes = [];
     const initialSchemesParams = [];
     const initialSchemesPermissions = [];
 
-    for (const optionScheme of options.schemes) {
-      /**
-       * TODO:  -dkent
-       * Note that at the moment this is only allowing Arc schemes, no non-Arc schemes.
-       * Can this change?
-       */
-      const arcSchemeInfo = contracts.allContracts[optionScheme.name];
+    for (const schemeOptions of options.schemes) {
+      const arcSchemeInfo = contracts.allContracts[schemeOptions.name];
       if (!arcSchemeInfo) {
         throw new Error(
           "Non-arc schemes are not currently supported here.  You can add them later in your workflow."
         );
       }
+
+      /**
+       * scheme will be a contract wrapper
+       */
       const scheme = await arcSchemeInfo.contract.at(
-        optionScheme.address || arcSchemeInfo.address
+        schemeOptions.address || arcSchemeInfo.address
       );
 
-      const paramsHash = await scheme.setParams({
-        voteParametersHash: voteParametersHash,
-        votingMachine: org.votingMachine.address,
-        orgNativeTokenFee: options.orgNativeTokenFee,
-        schemeNativeTokenFee: options.schemeNativeTokenFee
-      });
+      /**
+       * any supplied voting machine parameters for the scheme will override the global defaults
+       */
+      let schemeVotingMachineParams = {};
+      let schemeVoteParametersHash;
+
+      if (schemeOptions.votingMachineParams && (schemeOptions.votingMachineParams != {})) {
+
+        Object.assign(schemeVotingMachineParams, defaultVotingMachineParams, schemeOptions.votingMachineParams);
+
+        await dao.votingMachine.setParameters(
+          schemeVotingMachineParams.reputationAddress,
+          schemeVotingMachineParams.votePerc,
+          schemeVotingMachineParams.ownerVote
+        );
+        schemeVoteParametersHash = await dao.votingMachine.getParametersHash(
```

Comment:
this seems like something we could make easier in the future, e.g. setParameters could return the parameters hash

Reply:
That is exactly what I did with `setParams`.  I have been slowly going through and making this exact change.  I think I've caught all of them now.